SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/TouchGrid.swift

```diff
@@ -89,9 +89,20 @@ class TouchGrid {
             }
             
             if self.curNode.childNodes.count != 0 {
+                if(self.curNode.childNodes[0] as? ButtonNode)?.dummy==true{
```

Comment:
I'm pretty sure you can use exclamation points here, since we're sure the node will be a ButtonNode and will not be null. Not 100% sure though, so just see if you can replace the `?` with `!`. We want to use `!` as much as possible since it implies a certainty about our code. Also you don't need the `== true` in the `if` statement, since the `if` statement automatically asks if the expression is `== true`. Finally, please put a space after the `if`.

Reply:
Hi,  I tried to replace ? with ! but the program behave unnormally